src/en/kunmangato/src/eu/kanade/tachiyomi/extension/en/kunmangato/KunMangaTo.kt

```diff
@@ -0,0 +1,213 @@
+    override val baseUrl = "https://kunmanga.to"
+    override val lang = "en"
+    override val supportsLatest = true
+
+    private val json: Json by injectLazy()
+
+    private val dateFormat by lazy {
+        SimpleDateFormat("MMM d, yyyy", Locale.ENGLISH)
+    }
+
+    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val mangas = super.popularMangaParse(response).mangas.distinctBy { it.url }
+        return MangasPage(mangas, false)
+    }
+
+    override fun popularMangaSelector(): String = ".sidebar-box-popular article"
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        val a = element.selectFirst("a.manga")!!
+        return SManga.create().apply {
+            title = a.text()
+            url = a.attr("href").removePrefix(baseUrl)
+            thumbnail_url = element.selectFirst("img")!!.attr("src")
+        }
+    }
+
+    override fun popularMangaNextPageSelector(): String? = null
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val urlBuilder = "$baseUrl/search".toHttpUrl().newBuilder()
+            .addQueryParameter("term", query.trim())
+            .addQueryParameter("page", page.toString())
+
+        filters.filterIsInstance<UriPartFilter>().forEach { filter ->
+            urlBuilder.addQueryParameter(
+                filter.toQueryParam(),
+                filter.toUriPart(),
+            )
+        }
+
+        return GET(urlBuilder.build(), headers)
+    }
+
+    override fun searchMangaSelector(): String = "article .card-manga"
+
+    override fun searchMangaFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector(): String = "ul.pagination-primary a[rel=next]"
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val urlBuilder = "$baseUrl/latest".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+
+        return GET(urlBuilder.build(), headers)
+    }
+
+    override fun latestUpdatesSelector(): String = searchMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element) = searchMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector(): String = searchMangaNextPageSelector()
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            title = document.selectFirst(".page-heading")!!.text()
+            author = document.selectFirst("p.mb-1:nth-child(1)")!!.text().drop(9)
+            description = document.getElementById("manga-description")!!.text()
+            genre =
+                document.select("a.manga-genre").joinToString(", ") { element -> element.text() }
+            status =
+                if (document.select(".text-success").size > 0) SManga.COMPLETED else SManga.ONGOING
```

Comment:
This is somewhat flaky. Since it is so generic it will cause issues if the layout changes slightly. It would be preferable if it's `SManga.UNKOWN` by default, and only set to either completed or ongoing if found.

The document is a bit difficult to navigate, but you can rely on Jsoup's selectors:
```css
*:has(~ #manga-description) p span:contains(Status:) ~ span
```